daemon/tests/happy_path.rs

```diff
@@ -25,45 +26,25 @@ use std::time::Duration;
 use tokio::time::sleep;
 mod harness;
 
-/// Assert the next state of the single cfd present at both maker and taker
-macro_rules! assert_next_state {
-    ($id:expr, $maker:expr, $taker:expr, $state:expr) => {
-        // TODO: Allow fetching cfd with the specified id if there is more than
-        // one on the cfd feed
-        let (taker_cfd, maker_cfd) = next_cfd($taker.cfd_feed(), $maker.cfd_feed())
-            .await
-            .unwrap();
-        assert_eq!(
-            taker_cfd.order_id, maker_cfd.order_id,
-            "order id mismatch between maker and taker"
-        );
-        assert_eq!(
-            taker_cfd.state, maker_cfd.state,
-            "cfd state mismatch between maker and taker"
-        );
-        assert_eq!(taker_cfd.order_id, $id, "unexpected order id in the taker");
-        assert_eq!(maker_cfd.order_id, $id, "unexpected order id in the maker");
-        assert_eq!(taker_cfd.state, $state, "unexpected cfd state in the taker");
-        assert_eq!(maker_cfd.state, $state, "unexpected cfd state in the maker");
-    };
+/// Waits until the CFDs for both maker and taker are in the given state.
+macro_rules! wait_next_state {
```

Comment:
> this should not happen! What we call "brittle" here might actually be a flaw in the logic.

I don't think it is as clear cut to say that it shouldn't happen.

We cannot and IMO should not have control over ever single message that is sent in the system. That would require exposing way too many knobs and mocks. We have to accept that the actor system we are building are eventually consistent, meaning things will eventually be in the correct state if it works correctly.

> We _can_ change the test setup to wait for the next state and ignore "other" state changes in between, but that makes the test less "restrictive" in testing the behavior.

It would be fairly trivial to pass a list of states that it should go through before arriving at the final state. But outside of that, I don't think it is particularly likely for us to be in an unexpected state and then actually end up in the correct state in the end. And even if, is that really a bug or is that kind of self-healing functionality a feature? I don't think the user is gonna care much ...